cleanlab/rank.py

```diff
@@ -109,57 +123,335 @@ def get_self_confidence_for_each_label(
+    if verbose:
+        print(f"Weighting scheme for ensemble: {weight_ensemble_members_by}")
+
+        # Only print if the user specified to weight by accuracy
+        if weight_ensemble_members_by == "accuracy":
+            for i, acc in enumerate(accuracy_list):
+                print(f"Model {i} accuracy: {acc}")
+
+    # Transform list of scores into an array of shape (N, M) where M is the number of models in the ensemble
+    scores_ensemble = np.vstack(scores_list).T
+
+    # Aggregate scores with chosen weighting scheme
+    if weight_ensemble_members_by == "uniform":
+
+        # Uniform weights (simple average)
+        label_quality_scores = scores_ensemble.mean(axis=1)
+
+    elif weight_ensemble_members_by == "accuracy":
+
+        # Weight by accuracy
+        weights = np.array(accuracy_list) / sum(accuracy_list)
+
+        # Aggregate scores with weighted average
+        label_quality_scores = (scores_ensemble * weights).sum(axis=1)
+
+    else:
+        raise ValueError(
+            f"""
+            {weight_ensemble_members_by} is not a valid weighting method for weight_ensemble_members_by!
+            Please choose a valid weight_ensemble_members_by: uniform, accuracy
+            """
+        )
+
+    return label_quality_scores
+
+
+def get_self_confidence_for_each_label(
+    labels: np.array,
+    pred_probs: np.array,
+) -> np.array:
+    """Returns the self-confidence label-quality score for each datapoint.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
+    The self-confidence is the holdout probability that an example belongs to
+    its given class label.
+
+    Parameters
+    ----------
+    labels : np.array
+      Labels in the same format expected by the `score_label_quality()` method.
+
+    pred_probs : np.array (shape (N, K))
+      Predicted-probabilities in the same format expected by the `score_label_quality()` method.
+
+    Returns
+    -------
+    label_quality_scores : np.array (float)
+      Return the holdout probability that each example in pred_probs belongs to its
+      label.
+
+    """
 
     # np.mean is used so that this works for multi-labels (list of lists)
     label_quality_scores = np.array([np.mean(pred_probs[i, l]) for i, l in enumerate(labels)])
     return label_quality_scores
 
 
 def get_normalized_margin_for_each_label(
-    labels,
-    pred_probs,
-):
-    """Returns the "normalized margin" for every example associated pred_probs and
-    labels.
+    labels: np.array,
+    pred_probs: np.array,
+) -> np.array:
+    """Returns the "normalized margin" label-quality score for each datapoint.
+
+    This is a function to compute label-quality scores for classification datasets,
+    where lower scores indicate labels less likely to be correct.
+
     The normalized margin is (p(label = k) - max(p(label != k))), i.e. the probability
```

Comment:
Not your responsibility, but can we add:
"Letting `k` denote the given label for a datapoint, the normalized margin is ..."

Reply:
Done, thanks.
3dbbe7054aeae25b8bddd73a53838cce35b06fa9